### PR TITLE
Redis enterprise Integration- Beta package

### LINKED ISSUE #3457: Migrate Redis Enterprise Integration
# Integration release checklist

This checklist is intended for integrations maintainers to ensure consistency
when creating or updating a Package, Module or Dataset for an Integration.

### All changes

- [ ] Change follows the [contributing guidelines](https://github.com/elastic/integrations/blob/main/CONTRIBUTING.md)
- [ ] Supported versions of the monitoring target are documented
- [ ] Supported operating systems are documented (if applicable)
- [ ] Integration or [System tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/system_testing.md) exist
- [ ] Documentation exists
- [ ] Fields follow [ECS](https://github.com/elastic/ecs) and [naming conventions](https://www.elastic.co/guide/en/beats/devguide/master/event-conventions.html)
- [ ] At least a manual test with ES / Kibana / Agent has been performed.
- [ ] Required Kibana version set to:

### New Package

- [ ] Screenshot of the "Add Integration" page on Fleet added

### Dashboards changes

- [ ] Dashboards exists
- [ ] Screenshots added or updated
- [ ] Datastream filters added to visualizations

### Log dataset changes

- [ ] [Pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) exist (if applicable)
- [ ] Generated output for at least 1 log file exists

### Metric dataset changes

_This entry is currently only recommended. It will be mandatory once we provide better support for it._

- [ ] Sample event (`sample_event.json`) exists

### PR DESCRIPTION
**Pull request by lalit-satapathy**
## What does this PR do?

Redis Enterprise Integration beta release. Closes #3457

## Checklist

- [X] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [X] I have verified that all data streams collect metrics or logs.
- [X] I have added an entry to my package's `changelog.yml` file.
- [X] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Pending Items

- System test not planned refer ( [system-test-note.md](https://github.com/elastic/integrations/compare/main...lalit-satapathy:redis-enterprise-ver1?expand=1#diff-5742b68834ef31eb25b2f044b5885bfe7c3661aa8fe73988dd292d955471fa8e) )

## How to test this PR locally

- Docker install redislabs/redis:5.2.2-24
- Create a dummy node and database
- Ensure that the prometheus port (8070) is exported
- Pass the host:8070 to the integration hosts

## Related issues

- Closes #3457

## Screenshots

redisenterprise-dashboard

### CODE REVIEW

**Review by muthu-mps — COMMENTED** [review]
The metric names which appears inside the visualisation can be removed. Like `system, user, Free Memory` which is already part of the title.

File: packages/redisenterprise/data_stream/node/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+description: Pipeline for parsing Redis Enterprise metrics
+processors:
+  - set:
+      field: ecs.version
+      value: '8.2.0'
```
Comment (muthu-mps) [inline packages/redisenterprise/data_stream/node/elasticsearch/ingest_pipeline/default.yml:6]: The build.yml references ECS version 8.2.1. Do we need to make the version consistent in both the places ?

File: packages/redisenterprise/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+title: "Redis Enterprise"
+version: 0.1.1
+license: basic
+description: "Collect metrics from Redis Enterprise Cluster"
+type: integration
+categories:
+  - datastore
+release: beta
+conditions:
+  kibana.version: "^8.1.0"
```
Comment (ishleenk17) [inline packages/redisenterprise/manifest.yml:12]: We can have this from 8.2 onwards ?

File: packages/redisenterprise/data_stream/proxy/fields/fields.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  type: group
+  release: beta
+  fields:
+    - name: metrics
+      type: group
+      fields:
+        # use pattern for matching to ensure future extensibillity. 
+        - name: listener_*
+          type: float
+          metric_type: gauge
```
Comment (ishleenk17) [inline packages/redisenterprise/data_stream/proxy/fields/fields.yml:11]: IS it appropriate to say all the fields are gauge here ?

File: packages/redisenterprise/data_stream/node/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (lalit-satapathy) [reply packages/redisenterprise/data_stream/node/elasticsearch/ingest_pipeline/default.yml]: Updated.

File: packages/redisenterprise/manifest.yml (hunk 2 above)
Comment (lalit-satapathy) [reply packages/redisenterprise/manifest.yml]: Updated

File: packages/redisenterprise/data_stream/node/sample_event.json [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        "mac": [
+            "02:42:ac:15:00:07"
+        ],
+        "architecture": "x86_64"
+    },
+    "redisenterprise": {
+        "node": {
+            "metrics": {
+                "node_cpu_idle": 0.5973333333333333,
+                "node_ingress_bytes_max": 166.222,
```
[the remark this answers is not in the document]
Comment (lalit-satapathy) [reply packages/redisenterprise/data_stream/node/sample_event.json]: Fixed this using ingest pipeline and all node_ is removed.

File: packages/redisenterprise/data_stream/node/fields/fields.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  type: group
+  release: beta
+  fields:
+    - name: metrics
+      type: group
+      fields:
+        # use pattern for matching to ensure future extensibillity. 
+        - name: node_*
+          type: float
+          metric_type: gauge
```
[the remark this answers is not in the document]
Comment (lalit-satapathy) [reply packages/redisenterprise/data_stream/node/fields/fields.yml]: Removed gauge.

File: packages/redisenterprise/data_stream/proxy/sample_event.json [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        "mac": [
+            "02:42:ac:15:00:07"
+        ],
+        "architecture": "x86_64"
+    },
+    "redisenterprise": {
+        "proxy": {
+            "metrics": {
+                "listener_egress_bytes": 0,
+                "listener_last_req_time": 0,
```
[the remark this answers is not in the document]
Comment (lalit-satapathy) [reply packages/redisenterprise/data_stream/proxy/sample_event.json]: Yes, it depends on the db usage/load. These are prometheus metrics, so not an issue. Some values are there.

                "listener_last_res_time": 1657079277,

File: packages/redisenterprise/data_stream/proxy/fields/fields.yml (hunk 3 above)
Comment (lalit-satapathy) [reply packages/redisenterprise/data_stream/proxy/fields/fields.yml]: Removed gauge.

**Comment by lalit-satapathy** [comment]
> The metric names which appears inside the visualisation can be removed. Like `system, user, Free Memory` which is already part of the title.

Making it empty causes some issue, so we need some description there.

File: packages/redisenterprise/_dev/build/docs/README.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
+# Redis Enterprise
+
+Redis Enterprise integration provides monitoring of [redis](https://redis.com/) cluster. Monitoring is done via prometheus exported port of redis enterprise cluster. Once a redis enterprise [cluster](https://redis.com/redis-enterprise/technology/redis-enterprise-cluster-architecture/) is installed, prometheus port is available for monitoring. The url of the host:port(8070) needs to be passed to the hosts in the settings.
```
[the remark this answers is not in the document]
Comment (lalit-satapathy) [reply packages/redisenterprise/_dev/build/docs/README.md:3]: Typically 8070 is where the [prometheus](https://docs.redis.com/latest/rs/monitoring-metrics/prometheus-integration/) metrics are exported. The port reference in the doc will likely make it un-ambiguous for the user and easy to associate with.

File: packages/redisenterprise/data_stream/proxy/sample_event.json [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+        "version": "8.1.0"
+    },
+    "elastic_agent": {
+        "id": "24604ad6-3068-42da-8377-93b8c074e3c5",
+        "version": "8.1.0",
+        "snapshot": false
+    },
+    "@timestamp": "2022-07-06T03:53:58.638Z",
+    "ecs": {
+        "version": "8.2.0"
```
Comment (ishleenk17) [inline packages/redisenterprise/data_stream/proxy/sample_event.json:16]: The ecs version in sample event doesn't match the version in build.yml. 
Please regenerate the sample events.

Comment (ishleenk17) [reply packages/redisenterprise/data_stream/proxy/sample_event.json:16]: Please address this for both the datastreams

File: packages/redisenterprise/data_stream/node/fields/fields.yml (hunk 5 above)
[the remark this answers is not in the document]
Comment (ishleenk17) [reply packages/redisenterprise/data_stream/node/fields/fields.yml]: For now it is fine. But going forward we should see how we can handle such bulk assignments. 
Also, how having metric_type helps in the wider picture for better indexing.

File: packages/redisenterprise/data_stream/proxy/sample_event.json (hunk 8 above)
Comment (lalit-satapathy) [reply packages/redisenterprise/data_stream/proxy/sample_event.json]: regenerated sample events.

**Review by ishleenk17 — APPROVED** [review]

**Review by muthu-mps — APPROVED** [review]
LGTM!

**Pull request merged**